# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegistrationParent.h
# Commit: 185488c90cb8
# Full Hash: 185488c90cb8d3cd7f7041bc457f5e5cbb1617b2
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 15:47:45
# Regressor Bug: 1564235
# File Overlap Count: 2
# Description:
#   Bug 1564235 - P3 Create IPC for NavigationPreload in PServiceWorkerRegistration.ipdl. r=dom-worker-reviewers,asuth
#   
#   This patch implements the IPC for NavigationPreload APIs.
#   The IPC methods are created on PServiceWorkerRegistration.
#   
# ==============================================================================

diff -r afae3c50f7ac -r 185488c90cb8 dom/serviceworkers/ServiceWorkerRegistrationParent.h
--- a/dom/serviceworkers/ServiceWorkerRegistrationParent.h	Thu Jun 10 06:01:26 2021 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegistrationParent.h	Thu Jun 10 06:01:27 2021 +0000
@@ -33,6 +33,17 @@
   mozilla::ipc::IPCResult RecvUpdate(const nsCString& aNewestWorkerScriptUrl,
                                      UpdateResolver&& aResolver) override;
 
+  mozilla::ipc::IPCResult RecvSetNavigationPreloadEnabled(
+      const bool& aEnabled,
+      SetNavigationPreloadEnabledResolver&& aResolver) override;
+
+  mozilla::ipc::IPCResult RecvSetNavigationPreloadHeader(
+      const nsCString& aHeader,
+      SetNavigationPreloadHeaderResolver&& aResolver) override;
+
+  mozilla::ipc::IPCResult RecvGetNavigationPreloadState(
+      GetNavigationPreloadStateResolver&& aResolver) override;
+
  public:
   NS_INLINE_DECL_REFCOUNTING(ServiceWorkerRegistrationParent, override);
 